stephanie-cirne-s05/projeto_casa.py

```diff
@@ -0,0 +1,195 @@
+        if cod_opcao == 1: incluir_nova_aluna()
+        elif cod_opcao == 2: consultar_lista_alunas()
+        elif cod_opcao == 3: consultar_faltas_aluna()
+        elif cod_opcao == 4: consultar_notas_aluna()
+        elif cod_opcao == 5: consultar_status_aprovacao()
+        elif cod_opcao == 6: print("Encerrando o programa..."); break
+
+def obter_opcao():
+    codigo_opcao = 0
+
+    while codigo_opcao not in [1, 2, 3, 4, 5, 6]:
+        try:
+            codigo_opcao = int(input("\nEscolha uma opção:\n"
+                                    "1 - Incluir uma nova aluna\n"
+                                    "2 - Consultar lista de alunas\n"
+                                    "3 - Consultar faltas da aluna\n"
+                                    "4 - Consultar notas da aluna\n"
+                                    "5 - Consultar status de aprovação\n"
+                                    "6 - Sair do sistema\n"
+                                    "Opção: "))
+                
+            if codigo_opcao not in [1, 2, 3, 4, 5, 6]:
+                print("Opção inválida. Por favor, escolha uma opção válida (1 a 5).\n")
+        except ValueError:
+            print("Entrada inválida. Por favor, digite um número inteiro.\n")
+            
+        return codigo_opcao
+
+
+#Inserindo tupla com nome e sobrenome que será a chave principal no dicionário nova aluna   
+def incluir_nova_aluna():
+    inserir_nome_aluna = str (input ("Insira o primeiro nome da aluna: "))
+    inserir_sobrenome_aluna = str (input ("Insira o sobrenome da aluna: "))
+    nova_aluna = (inserir_nome_aluna, inserir_sobrenome_aluna)
+    dataset [nova_aluna] = {}
+
+   
+
+    turma = incluir_turma (nova_aluna)
+    notas = lista_notas (nova_aluna)
+    presenca = lista_presenca(nova_aluna)
+    nota_particpacao = particpacao (nova_aluna)
+  
+
+    salva_dados (nova_aluna,turma, notas, presenca, nota_particpacao)
+
+
+    return nova_aluna
+
+
+def incluir_turma (nova_aluna):
+    print (f"A nova aluna {nova_aluna} foi incluída com sucesso \n")
+
+    turma = str (input ("Insira a turma da aluna: "))
+    dataset [(nova_aluna)] = [turma]
+
+    
+
+    print (f"A {turma} foi incluída no cadastro da aluna {nova_aluna}")
+
+    return turma
+
+def lista_notas (nova_aluna):
+
+    print ("Insira uma nota de 1 a 10 cada matéria: \n")
+    
+    lista_notas = []
+
+    for i in range (1,4):
+        valor = float (input (f"Insira uma nota de 1 a 10 para a matéria {i}: \n"))
+        lista_notas. append (valor)
+
+    dataset [(nova_aluna)] = [lista_notas]
+
+    print (f"As notas {lista_notas} foram incluídas no sistema! \n")
+
+    return lista_notas
+
+
+def lista_presenca (nova_aluna):
+
+    lista_presenca_2 =[] 
+
+    for i in range (1,6):
+        valor = input(f"A aluna esteve presnete na aula {i}? Insira 'True' para presença e 'False' para ausência em cada aula:  \n")
+        lista_presenca_2.append (valor)
+
+    dataset [(nova_aluna)] = [lista_presenca_2] 
```

Comment:
Neste ponto, assim como nas outras duas funções, você está substituindo os dados que estão no dicionário para a chave do nome da aluna por uma lista com lista de presença, então ficaria:
de: [[9.0, 9.0, 9.0]]
para: [['True', 'True', 'True', 'True', 'True']]
Se quiséssemos atribuir neste ponto, poderiamos usar desta forma:

**Sugestão:**
Ou removemos esta linha e atribuimos os dados de uma só vez no salva_dados, ou em cada função adicionamos os dados no dicionário desta forma:
```suggestion
    dataset[nova_aluna]["Presença"] = lista_presenca_2
```